lib/elixir/src/elixir_expand.erl

```diff
@@ -1116,7 +1116,10 @@ format_error({invalid_function_call, Expr}) ->
 format_error({invalid_call, Call}) ->
   io_lib:format("invalid call ~ts", ['Elixir.Macro':to_string(Call)]);
 format_error({invalid_quoted_expr, Expr}) ->
-  io_lib:format("invalid quoted expression: ~ts", ['Elixir.Kernel':inspect(Expr, [])]);
+  Message =
+    "invalid quoted expression: ~ts\n"
+    "You may have meant to escape this with Macro.escape/1",
```

Comment:
I think we can be a bit more descriptive. Something like this:

```
"invalid quoted expression: %{foo: :bar}\n\n"
"Please make sure your quoted expressions are made of valid AST nodes."
"If you would like to introduce a value into the AST, such as a four-element"
tuple or a map, make sure to call `Macro.escape/1` before"
```

Thoughts?

Reply:
Sounds good to me! I'll update the branch.